### DISCUSSION TITLE
Versionvaihdon tiedote 4 : Versionvaihto alkaa Kohan tuotantoympäristöissä

### DISCUSSION THREAD

**Discussion by CodoDerDritte** [KohaSuomi/Koha#536] [Tiedotteet]
Aloitamme Kohan tuotantoympäristöjen versionvaihtotyöt tänään 15.5. klo 22.00. Tuolloin siis kaikki tuotantokohat menevät alas päivitystä varten. Suunniteltu työjärjestys ja versionvaihtoon liittyvät muistiinpanot löytyvät [Versiovaihdon muistiinpanot](https://github.com/KohaSuomi/Koha-22x/wiki/Versiovaihdon-muistiinpanot) -dokumentista.

Tiedotamme tässä keskusteluketjussa vaihtotyön etenemisestä.

**Comment by CodoDerDritte** [comment]
Konfiguraatiotiedostot on päivitetty ja ajastukset pysäytetty. Tuotantokohat menevät alas pian.

**Comment by CodoDerDritte** [comment]
Tuotantoihin vaihdetaan nyt uusi Koha pohjalle ja sen jälkeen käynnistetään tuotantokontainerit. Sen jälkeen tietokantapäivityksiä.

**Comment by CodoDerDritte** [comment]
Tietokantapäivityksiä ja uudelleenindeksointeja tehdään parhaillaan. Niiden pitäisi olla pääosin valmiina huomenna aamulla. Pääsy tuotanto-Kohiin on estetty, jottei kukaan kirjaudu sinne ennen aikojaan. Tällä hetkellä tarjoilevat siis Service unavailablea ihan tarkoituksella.

**Comment by emta001** [comment]
Tietokantapäivitykset on nyt ajettu. Myös systempreferences- ja letter-taulut on kopioitu nexteiltä tuotantoihin.

**Comment by emta001** [comment]
Uudelleenindeksoinnit jatkuvat vielä Vaskissa, Hellessä ja Lapissa. Siilin, Lumpeen, Kyytin ja Vaaran indeksoinnit ovat valmistuneet yön aikana. OUTIssa indeksointi oli kaatunut puuttuvaan asetustiedostoon. Tämä on nyt korjattu ja uudelleenindeksointi on saatu myös OUTIn osalta käyntiin.

**Comment by CodoDerDritte** [comment]
Indeksoinnit ovat Outia ja Tätiä lukuunottamatta valmiit.

**Comment by CodoDerDritte** [comment]
Maksut on laskettu ja asiakasviestejä generoidaan parhaillaan. SIP-palvelimet on myös käynnistetty osassa kimppoja.

**Comment by CodoDerDritte** [comment]
Käynnistämme seuraavaksi ajastetut ajot. Siinä lähtee käyntiin asiakasviestien lähetys, hyllyvarausten- ja hankintasanomien käsittely ym.

**Comment by CodoDerDritte** [comment]
Ajastukset käynnissä. Sähköpostit ja tekstiviestit otettiin vielä hetkeksi pois käytöstä, etteivät asiakkaat turhaan tule repimään suljetun kirjaston ovea viestin saatuaan. Noutoilmoituksia saattoi kyllä livahtaa jo tuossa 15.30 maaillmalle ennenkuin ehdittiin ottaa viestien lähetys pois käytöstä. Kirjeviestit lähtevät illalla normaaliaikataulun mukaisesti.

**Comment by CodoDerDritte** [comment]
Päivitys on nyt tämän päivän osalta pulkassa eikä ilmeisesti mitään vakavia ongelmia ole ilmennyt. Pistämme pillit pussiin tältä päivältä ja jatkamme pikku detaljien hiomista huomenna.

**Comment by CodoDerDritte** [comment]
Ajastettujen tehtävien ajamisessa on ollut ongelmia, koska niihin liittyvät ajastusskriptit oli linkittämättä. Nämä on nyt korjattu ja ajot menevät (toivottavasti) oikein ensi yönä. Finnat on avattu kaikissa kimpoissa ja asiakasviestintä rullaa normaalisti Outissa, Siilinjärvessä, Kyytissä ja Vaskissa.

**Comment by CodoDerDritte** [comment]
Vaskin ja Lumpeen asiakasviestintä käynnistetään perjantaina aamulla. Lapista ei ole vielä saatu tietoa milloin halutaan käynnistää. Kaikenlaista pikkupaikkailua on tänään tehty sinne tänne, kuten oli odotettavissakin.

Joidenkin automaattien kanssa on havaittu ongelmia jos Kohassa jollekkin asiakastyypille on ruksittuna asiakastyypin asetuksissa "Can be a quarantee" ja tällä asiakastyypillä on ollut asiakkaita, joilla ei ole takaajaa. Tyypillisesti nämä ovat olleet kirjastojen virkailijatunnuksia tai muita ei-asiakastunnuksia. Automaattiongelmien tullessa vastaan tämä lienee ensimmäinen asia, joka kannatta tarkistaa.

Koha-Suomi siirtyy pikkuhiljaa huilailemaan helatorstain ajaksi ja jatkaa uudella puhdilla varustettuna taas perjantaina. Mukavaa vapaapäivää itse kullekkin ja kiitos ahkerasta testailusta ja kärsivällisyydestä :)

**Comment by CodoDerDritte** [comment]
Loppujen kimppojen asiakasviestintä on käynnistetty tänään aamulla. Versiopäivitys on itse päivityksen osalta taputeltu, joskin toki työ vielä toimimattomien ominaisuuksien parissa tietysti jatkuu. Seuraava korjauspäivitys tulee normaaliaikataulun mukaisesti ensi tiistaina aamulla. Lopetamme tiedottamisen tässä ketjussa tähän ja hommat jatkuvat sitten omien tikettiensä alla.
